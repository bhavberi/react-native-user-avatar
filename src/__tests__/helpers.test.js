/* eslint max-len: 0 */

import {
  abbr,
  sumChars,
  fetchImage,
  generateBackgroundStyle,
  generateBackgroundColor,
  getContainerStyle,
} from "../helpers";

beforeEach(function() {
  global.fetch = jest.fn().mockImplementation(() => {
    mockResponseObject = {
      _bodyBlob: {
        _data: {},
      },
      _bodyInit: {
        _data: {},
      },
      headers: {
        map: {
          "cache-control": "public, max-age=31536000",
          "connection": "keep-alive",
          "content-length": "271",
          "content-type": "image/png",
          "date": "Thu, 15 Aug 2024 18:10:00 GMT",
          "last-modified": "Tue, 13 Aug 2024 21:08:11 GMT",
          "server": "cloudflare",
          "x-powered-by": "WordOps",
        },
      },
      ok: true,
      status: 200,
      url: "https://dummyimage.com/100x100/000/fff",
    };

    const p = new Promise((resolve, reject) => {
      resolve(mockResponseObject);
    });
    return p;
  });
});

describe("test abbr function normally", () => {
  it("should return my initials", () => {
    expect(abbr("Bhav Beri")).toMatchSnapshot();
  });
});

describe("test abbr function when name starts with +", () => {
  it("should return my initials", () => {
    expect(abbr("+Bhav Beri")).toMatchSnapshot();
  });
});

describe("test abbr function when name is invalid", () => {
  it("should return my initials", () => {
    expect(abbr("!@# #!$#")).toMatchSnapshot();
  });
});

describe("test sumChars function", () => {
  it("should return the same sum", () => {
    expect(sumChars("Bhav Beri")).toMatchSnapshot();
  });
});

describe("test fetchImage function on an non-image URL", () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() => {
      mockResponseObject = {
        _bodyBlob: {
          _data: {},
        },
        _bodyInit: {
          _data: {},
        },
        headers: {
          map: {
            "connection": "keep-alive",
            "content-type": "text/html",
            "date": "Thu, 15 Aug 2024 18:10:00 GMT",
            "last-modified": "Tue, 13 Aug 2024 21:08:11 GMT",
            "server": "Github.com",
          },
        },
        ok: true,
        status: 200,
        url: "https://github.com/bhavberi/react-native-user-avatar/",
      };

      const p = new Promise((resolve, reject) => {
        resolve(mockResponseObject);
      });
      return p;
    });
  });

  it("calls the non-image url and returns false", async () => {
    jest.useFakeTimers();
    fetchImage("https://github.com/bhavberi/react-native-user-avatar/").then(
      (res) => {
        expect(res).toBe(false);
      },
    );
  });

  it("calls the non-image url with ignoreImageTypeCheck and returns true", async () => {
    jest.useFakeTimers();
    fetchImage(
      "https://github.com/bhavberi/react-native-user-avatar/",
      {},
      true,
    ).then((res) => {
      expect(res).toBe(true);
    });
  });
});

describe("test fetchImage function on an invalid URL", () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation(() => {
      mockResponseObject = {
        _bodyBlob: {
          _data: {},
        },
        _bodyInit: {
          _data: {},
        },
        headers: {
          map: {
            "connection": "keep-alive",
            "content-type": "text/html",
            "date": "Thu, 15 Aug 2024 18:10:00 GMT",
            "last-modified": "Tue, 13 Aug 2024 21:08:11 GMT",
            "server": "Github.com",
          },
        },
        ok: false,
        status: 404,
        url: "https://github.com/bhavberi/react-native-user-avatara/",
      };

      const p = new Promise((resolve, reject) => {
        resolve(mockResponseObject);
      });
      return p;
    });
  });

  it("calls the invalid image url and returns false", async () => {
    jest.useFakeTimers();
    fetchImage("https://github.com/bhavberi/react-native-user-avatara/").then(
      (res) => {
        expect(res).toBe(false);
      },
    );
  });
});

describe("test generateBackgroundStyle function", () => {
  it("should return the same sum", () => {
    expect(
      generateBackgroundStyle("Bhav Beri", null, [
        "#2ecc71",
        "#3498db",
        "#8e44ad",
        "#e67e22",
        "#e74c3c",
        "#1abc9c",
        "#2c3e50",
      ]),
    ).toMatchSnapshot();
  });
});

describe("test generateBackgroundStyle function", () => {
  it("should return the same sum", () => {
    expect(
      generateBackgroundStyle("Bhav Beri", "#2ecc71", [
        "#2ecc71",
        "#3498db",
        "#8e44ad",
        "#e67e22",
        "#e74c3c",
        "#1abc9c",
        "#2c3e50",
      ]),
    ).toMatchSnapshot();
  });
});

describe("test generateBackgroundColor function", () => {
  it("should return the same sum", () => {
    expect(
      generateBackgroundColor("Bhav Beri", null, [
        "#2ecc71",
        "#3498db",
        "#8e44ad",
        "#e67e22",
        "#e74c3c",
        "#1abc9c",
        "#2c3e50",
      ]),
    ).toMatchSnapshot();
  });
});

describe("test generateBackgroundColor function", () => {
  it("should return the same sum", () => {
    expect(
      generateBackgroundColor("Bhav Beri", "#2ecc71", [
        "#2ecc71",
        "#3498db",
        "#8e44ad",
        "#e67e22",
        "#e74c3c",
        "#1abc9c",
        "#2c3e50",
      ]),
    ).toMatchSnapshot();
  });
});

describe("test getContainerStyle function", () => {
  it("should return the same sum", () => {
    jest.useFakeTimers();
    expect(
      getContainerStyle(32, "https://dummyimage.com/100x100/000/fff", 64),
    ).toMatchSnapshot();
  });
});
