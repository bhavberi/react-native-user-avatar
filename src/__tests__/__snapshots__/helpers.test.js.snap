// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test abbr function normally should return my initials 1`] = `"BB"`;

exports[`test abbr function when name is invalid should return my initials 1`] = `"!@"`;

exports[`test abbr function when name starts with + should return my initials 1`] = `"+B"`;

exports[`test generateBackgroundColor function should return the same sum 1`] = `"#1abc9c"`;

exports[`test generateBackgroundColor function should return the same sum 2`] = `"#2ecc71"`;

exports[`test generateBackgroundStyle function should return the same sum 1`] = `
{
  "backgroundColor": "#1abc9c",
}
`;

exports[`test generateBackgroundStyle function should return the same sum 2`] = `
{
  "backgroundColor": "#2ecc71",
}
`;

exports[`test getContainerStyle function should return the same sum 1`] = `
{
  "alignItems": "center",
  "borderColor": "transparent",
  "borderRadius": 64,
  "borderWidth": 0,
  "justifyContent": "center",
}
`;

exports[`test sumChars function should return the same sum 1`] = `803`;
